xc/device.ml

```diff
@@ -1986,6 +1986,11 @@ module Backend = struct
               )
 
           let qmp_event_thread () =
+            (* Add the existing qmp sockets first *)
```

Comment:
This can stay as it is. A different way of writing this would be:
```
Sys.readdir var_run_xen_path
|> Array.to_list
|> List.iter (fun x -> try Scanf.sscanf x "qmp-event-%d" add with _ -> ())
```
The advantage of this is that it avoids parentheses and the flow of data is simply from top to bottom. But this is a matter of taste.

Reply:
Thanks, I change it as below
```
+            try
+              Sys.readdir var_run_xen_path
+              |> Array.to_list
+              |> List.iter (fun x -> try Scanf.sscanf x "qmp-event-%d" add with _ -> ())
+            with e -> error "Failed to add qmp event socket: %s" (Printexc.to_string e);
```
Because xenopsd will execute `./xenopsd-xc --help=groff > xenopsd-xc.1` to generate man file which executes the qmp thread. In build envrionment,  directory `/var/run/xen` doesn't exist, it will shows below error:
```
./xenopsd-xc --help=groff > xenopsd-xc.1
Thread 2 killed on uncaught exception Sys_error("/var/run/xen: No such file or directory")
Raised by primitive operation at file "xc/device.ml", line 1992, characters 29-57
Called from file "thread.ml", line 39, characters 8-14
```

Reply:
This suggests that xenopsd is entering into a daemon mode when it really should not because it is only emitting some output in response to `--help` and not serving as a daemon.

Reply:
That's because `_init_qmp_event` is a static value that creates the qmp thread. Perhaps we should make `_init_qmp_event` a function and call it from where the xenopsd daemon is supposed to initialise?

Reply:
Yes.

Reply:
The xenopsd daemon initialisation for the libxc/upstream-qemu backend is made inside `xenops_server_xen.init()`, so we probably should call `Device.Backend.Qemu_upstream.Dm.QMP_Event.init()` from inside `xenops_server_xen.init()` in the line after initialising the xenstore watcher in `Watcher.create_watcher_thread()` .

Reply:
Cool, thanks. Updated.